### DISCUSSION TITLE
Quasar + Vuex 4 + Firebase 9

### DISCUSSION THREAD

**Discussion by Revaycolizer** [quasarframework/quasar#14738] [General - Components / Directives / etc]
How to fetch data from Firebase 9 through Vuex 4

**Comment by MilosPaunovic** [comment]
What have you tried so far and where does Quasar causing issues?

**Reply by Revaycolizer** [reply]
I am just a beginner sir i was wondering working with composition API as the firebase docs can be used in option API i wonder how to implement the method to save and fetch data

**Reply by MilosPaunovic** [reply]
If you are a beginner, the best way for you to learn is to start doing stuff by yourself. So, hop on Google, try searching for what you need and try to implement it. If you stumble upon a Quasar related, specific problem, then this is the place to ask.

For starters, you can check this out: https://www.section.io/engineering-education/quasar-googleauth-with-firebase/
and search through the discussions on Quasar repo to see if anything there could help you.

**Comment by Revaycolizer** [comment]
I am having trouble saving wnd fetching data from firebase through vuex
store

**Comment by suleimanas** [comment]
[vuex using vite](https://quasar.dev/quasar-cli-vite/state-management-with-vuex#introduction)
[vuex using webpack](https://quasar.dev/quasar-cli-webpack/state-management-with-vuex#introduction)
- or -
[pinia using vite](https://quasar.dev/quasar-cli-vite/state-management-with-pinia)
[pinia using webpack](https://quasar.dev/quasar-cli-webpack/state-management-with-pinia)
